haystack/components/evaluators/results_evaluator.py

```diff
@@ -0,0 +1,58 @@
+from typing import Any, Dict
+
+from pandas import DataFrame
+from pandas import concat as pd_concat
+
+
+class EvaluationResults:
+    def __init__(self, pipeline_name: str, results: Dict[str, Any]):
+        self.results = results
+        self.pipeline_name = pipeline_name
+
+    def score_report(self) -> DataFrame:
+        """Calculate the average of the scores for each metric."""
+        results = {entry["name"]: sum(entry["scores"]) / len(entry["scores"]) for entry in self.results["metrics"]}
+        return DataFrame.from_dict(results, orient="index", columns=["score"])
+
+    def to_pandas(self) -> DataFrame:
+        """
+        Creates a DataFrame containing the scores for each query and each metric.
+
+        :return: A DataFrame with the scores.
+        """
+        inputs_columns = list(self.results["inputs"].keys())
+        inputs_values = list(self.results["inputs"].values())
+        inputs_values = list(map(list, zip(*inputs_values)))  # transpose the values
+        df_inputs = DataFrame(inputs_values, columns=inputs_columns)
+
+        scores_columns = [entry["name"] for entry in self.results["metrics"]]
+        scores_values = [entry["scores"] for entry in self.results["metrics"]]
+        scores_values = list(map(list, zip(*scores_values)))  # transpose the values
+        df_scores = DataFrame(scores_values, columns=scores_columns)
+
+        return df_inputs.join(df_scores)
+
+    def comparative_individual_score_report(self, other: "EvaluationResults") -> DataFrame:
```

Comment:
need to align with @julian-risch on the names for these for consistency

Reply:
The names had already been discussed on the proposal: https://github.com/deepset-ai/haystack/pull/7462/files

I just renamed them according to your suggestions in your previous comments.